admin_guide/managing_networking.adoc

```diff
@@ -1201,3 +1201,57 @@ objects:
             name: default
 ...
 ----
+
+[[admin-guide-enabling-hsts]]
+== Enabling HTTP Strict Transport Security
+
+HTTP Strict Transport Security (HSTS) policy is a security enhancement, which
+requests that traffic to the host be HTTPS. This is useful for ensuring secure
+interactions with websites, or to offer secure application for the user's
+benefit.
+
+The default behavior is to drop any HTTP requests that require HTTPS. Enabling
+HSTS lets you set up a redirect so that the first time a client requests the
+target it is redirected to HTTPS. Any subsequent attempts will get a HTTPS
+header in the response, meaning future requests will be HTTPS.
+
+HSTS adds a Strict Transport Security header to HTTPS responses from the site.
+The client (often a browser) reads the header and, from then on, can enforce the
+policy. All requests to the HTTP URL are changed to HTTPS before the request is
+sent, eliminating the need for a redirect.
+
+[IMPORTANT]
+====
+HSTS works only with secure routes (either edge terminated or re-encrypt). The
+configuration is ineffective on HTTP or passthrough routes.
+====
+
+To enable HSTS to a route, add the `haproxy.router.openshift.io/hsts_header`
+value to the route object:
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Route
+metadata:
+  annotations:
+    haproxy.router.openshift.io/hsts_header: max-age=31536000;includeSubDomains;preload
+----
+
+
+The required `max-age` parameter indicates the length of time, in seconds, the
+HSTS policy is in effect for. The client updates `max-age` whenever a response
+with a HSTS header is received from the host. When `max-age` times out, the
+client discards the policy. 
+
+[NOTE]
+====
+To disable HSTS on a route, set `max-age=0`. Do not delete the annotation or
+route.
+====
+
+The optional `includeSubDomains` parameter tells the client that all subdomains
+of the host are to be treated the same as the host.
+
+If `max-age` is greater than 0, the optional `preload` parameter tells the
```

Comment:
Preload tells external services that they can include this site in their HSTS preload lists.  https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security#Preloading_Strict_Transport_Security

So, google keeps track of who has this set, and then builds a list of sites.  Browsers can consume these lists to work out what sites to only talk to over HTTPS without ever having talked to the site before.  Without preload, they need to have talked to the site over https to get the header... an attacker could theoretically proxy a client's traffic to the site to http and strip the header so that the https upgrade never happened.

Reply:
Also, clients often include a compiled in preload list.